docs/additional/interesting_aspects_de.md

```diff
@@ -0,0 +1,248 @@
+
+[docker-multi-stage]: https://docs.docker.com/develop/develop-images/multistage-build/
+
+### CES-freie Entwicklungsumgebung
+
+Das Bauen des Dogus und das Ausliefern ins lokale EcoSystem fordert nicht nur viel Zeit, sondern nimmt auch die Möglichkeit die Software effektiv zu debuggen.  
+Daher ist es sinnvoll eine lokale Entwicklungsumgebung unabhängig vom CES aufzubauen, um die Dogu-Software effektiv auszuführen, zu testen und zu debuggen.
+
+Die lokale Entwicklungsumgebung muss alle abhängigen CES-Dienste durch lokal gestartete Alternativen zu Verfügung stellen.
+Wir empfehlen, dass eine lokale Entwicklungsumgebung in Form einer `docker-compose.yml`-Datei definiert ist und jederzeit einfach mit `docker-compose up` gestartet und mit `docker-compose down` heruntergefahren werden kann.  
+Für ein Beispiel [siehe CAS][local-cas-example].
+
+[local-cas-example]: https://github.com/cloudogu/cas/blob/develop/app/docker-compose.yml
+
+## Qualitätssicherung für Dogus
+
+### Container Validation (GOSS)
+
+Die Container-Validation versichert die richtige Konfiguration des Containers, die für eine reibungslose Ausführung der Software notwendig ist.  
+
+Dafür verwenden wir in unseren Dogus das Server-Validation Programm [goss][goss].  
+Die eigentlichen goss-Tests werden in der folgenden Datenstruktur im Dogu angelegt:
+- `<Root-Verzeichnis des Dogus>/`
+  - `spec/`
+    - `goss/`
+      - `goss.yml` (enthält alle goss-Tests)
+
+Die goss-Tests sollten am besten alle wichtigen Aspekte für den reibungslosen Start eines Dogus überprüfen.  
+Darunter fallen Aspekte wie:
+- Überprüfen der Berechtigung für Skripte (startup.sh), Volumes
+- Überprüfen der UID & GID für Dateien (startup.sh, resources, Volumes, Schreib-abhängige Pfade)
+- Erreichbarkeit der Software: TCP-Health-Checks
+
+Für die Integration in eine Pipeline empfehlen wir den Einsatz der `ecoSystem.verify()` Methode aus der [dogu-build-lib][dogu-build-lib].  
+Ein beispielhafter Aufruf sieht in einem Jenkinsfile wie folgt aus:
+
+```groovy
+stage('Verify') {
+    ecoSystem.verify("/dogu")
+}
+```
+
+Dieser ruft intern den Befehl `cesapp verify` auf, welcher die Goss-Tests ausführt.
+
+[goss]: https://github.com/goss-org/goss
+[dogu-build-lib]: https://github.com/cloudogu/dogu-build-lib
+
+## Dokumentation
+
+Jedes Tool sollte eine sinnvolle Dokumentation über verwendete Features besitzen, die während der Benutzung oder Support unterstützt.
+Um eine einfache Integration in unsere vorhandene Dokumentations-Infrastruktur zu ermöglichen, sollte die Dokumentation in einem im Repository-Root befindlichen Order `/docs` abgelegt werden.
+
+Standardmäßig umfasst die Doku folgende Dinge:
+- Features von dem Dogu
+- Installation des Dogus
+- Konfiguration des Dogus
+- Entwicklung am Dogu
+- Readme
+- Changelog
+
+### Lizenz
+
+Die Nutzungslizenz des Dogus sollte deutlich hervorgehen.
+Diese kann zum Beispiel in der Readme angegeben werden.
```

Comment:
It would also help to provide a full license file if one exists, f.i. in case of a publicly available repository (it does not have to be a FOSS repository)